service/manager.go

```diff
@@ -426,24 +430,28 @@ func (m *manager) startModules() []error {
 
 func (m *manager) stopModules() []error {
 	var results []error
-	var lock sync.Mutex
-	wg := sync.WaitGroup{}
-	wg.Add(len(m.moduleWrappers))
 	for _, mod := range m.moduleWrappers {
-		go func(m *moduleWrapper) {
-			if !m.IsRunning() {
-				// TODO: have a timeout here so a bad shutdown
-				// doesn't block everyone
-				if err := m.Stop(); err != nil {
-					lock.Lock()
-					results = append(results, err)
-					lock.Unlock()
-				}
+		errC := make(chan error, 1)
+		go func(mod *moduleWrapper) {
+			if mod.IsRunning() {
+				errC <- mod.Stop()
+				return
 			}
-			wg.Done()
+			errC <- nil
 		}(mod)
+
+		select {
+		case err := <-errC:
+			if err != nil {
+				results = append(results,
```

Comment:
Why? Select chooses one channel to receive and we append results slice only once per iteration.